### PR TITLE
Feat: Create type for Timeline in TypeScript

### PR DESCRIPTION
**Pull request by tsm1820**
# Notion Task

https://www.notion.so/70204328dfbc4abfa76a39d9c3b50169?v=2db6790b04df468cb2f1e85a6c117567&p=b9bec7deee434f4c8a0d92c274b88889

# Changes Made In This PR

Added a new ts file called Timeline.ts where we added types such as User and Timeline.

# Also In This PR

No other changes.

# Screenshots

No UI changes.

### CODE REVIEW

File: src/types/Timeline.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export type User = {
+  userId: string;
+  name: string;
+  timeline: string;
+};
```
Comment (millasml) [inline src/types/Timeline.ts:5]: we dont create a new User type. Insead, add the `timeline` field to the type in `User.ts`

File: src/types/Timeline.ts [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+export type Timeline = {
+  timelineId: string;
```
Comment (millasml) [inline src/types/Timeline.ts:8]: just `id` would do

**Review by millasml — CHANGES_REQUESTED** [review]
some tweaks

**Comment by millasml** [comment]
next time, remember to request a review from me!

File: src/types/Timeline.ts (hunk 1 above)
Comment (tsm1820) [reply src/types/Timeline.ts]: Thank you for the comments Milla! Added the timeline field to User.ts

File: src/types/Timeline.ts (hunk 2 above)
Comment (tsm1820) [reply src/types/Timeline.ts]: okay, changed to id :)

File: src/types/User.ts [hunk 3]
```diff
@@ -3,6 +3,7 @@ export interface User {
   uid: string;
   displayName?: string;
   email?: string;
+  timeline: string;
```
Comment (millasml) [inline src/types/User.ts:6]: make this `timelineId`

File: src/types/Timeline.ts [hunk 4]
```diff
@@ -0,0 +1,11 @@
+export type Timeline = {
+  id: string;
+  activities: {
+    isNewEntry: boolean;
+    resourceId: string;
+    date: number;
+    title: string;
+    description: string;
+    type: string;
+  };
```
Comment (millasml) [inline src/types/Timeline.ts:10]: sorry i think i missed this the previous review, but I don't think this is how you meant for the type to be. What you want is a bunch of activities with a unique id as key. Also, the journals have something called activities too so lets name this "entries". What you should do is first, create the `Entry` type separately:

```
export type Entry = {
    isNew: boolean;
    resourceId: string;
    date: number;
    title: string;
    description: string;
    type: string;
}
```

and then the timeline type should be
```
export type Timeline = {
   id: string;
   entries: { [entryId: string] : Entry}
}
}
```

Comment (millasml) [reply src/types/Timeline.ts:10]: you can read more about index signatures here: https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures

**Review by millasml — CHANGES_REQUESTED** [review]
sorry i missed some out!

File: src/types/Timeline.ts (hunk 4 above)
Comment (tsm1820) [reply src/types/Timeline.ts]: okay, i get it now, thank you!! just pushed the latest changes :)

**Review by millasml — APPROVED** [review]
LGTM! 

do note to use more descriptive commit messages

**Pull request merged**